subiquity/server/tests/test_server.py

```diff
@@ -96,8 +101,56 @@ def test_bogus_autoinstall_argument(self):
         with self.assertRaises(Exception):
             self.server.select_autoinstall()
 
+    async def test_wait_for_cloudinit_dryrun_sets_cloud_init_ok(self):
+        self.server.controllers = mock.Mock()
+        self.server.controllers.instances = []
+        self.assertIsNone(self.server.cloud_init_ok)
+        self.server.opts.dry_run = True
+        await self.server.wait_for_cloudinit()
```

Comment:
This really asks the parent system if cloud-init is OK, I suppose we should mock that out.

Reply:
oops missed that mock once I did it on the other test. Fixing now

Reply:
added an `self.assertEqual(0, wait_for.call_count)` for this test